src/compute-client/src/controller/replica.rs

```diff
@@ -306,26 +277,6 @@ where
         );
 
         self.metrics.inner.command_queue_size.dec();
-
-        // Initialize or drop per-collection state.
-        match command {
-            ComputeCommand::CreateInstance(config) => {
-                for &id in config.logging.index_logs.values() {
-                    let as_of = Antichain::from_elem(T::minimum());
-                    self.add_collection(id, as_of);
-                }
-            }
```

Comment:
I might be missing where exactly we still ensure that each of `index_logs` will have `add_collection` called for them. Is that now a consequence of some code in Adapter that uses the compute controller's API?

Reply:
The `ReplicaTask::add_collection` doesn't exist anymore, but it is equivalent to the new `ReplicaState::add_collection`. This is called for log collections whenever a new replica is added:

* `ActiveInstance::add_replica` calls `Instance::add_replica_state` for the new replica
* `Instance::add_replica_state` calls `ReplicaState::add_collection` for each existing collection on the instance, which includes the log collections